jolt-core/src/jolt/instruction/divu.rs

```diff
@@ -0,0 +1,264 @@
+use common::constants::virtual_register_index;
+use tracer::{ELFInstruction, RVTraceRow, RegisterState, RV32IM};
+
+use super::VirtualInstructionSequence;
+use crate::jolt::instruction::{
+    add::ADDInstruction, mulu::MULUInstruction, virtual_advice::ADVICEInstruction,
+    virtual_assert_eq_signs::ASSERTEQSIGNSInstruction,
+    virtual_assert_lt_abs::ASSERTLTABSInstruction, virtual_assert_lte::ASSERTLTEInstruction,
+    JoltInstruction,
+};
+/// Perform signed*unsigned multiplication and return the upper WORD_SIZE bits
+pub struct DIVUInstruction<const WORD_SIZE: usize>;
+
+impl<const WORD_SIZE: usize> VirtualInstructionSequence for DIVUInstruction<WORD_SIZE> {
+    fn virtual_sequence(trace_row: RVTraceRow) -> Vec<RVTraceRow> {
+        assert_eq!(trace_row.instruction.opcode, RV32IM::DIVU);
+        // DIVU operands
+        let x = trace_row.register_state.rs1_val.unwrap();
+        let y = trace_row.register_state.rs2_val.unwrap();
+        // DIVU source registers
+        let r_x = trace_row.instruction.rs1;
+        let r_y = trace_row.instruction.rs2;
+        // Virtual registers used in sequence
+        let v_0 = Some(virtual_register_index(0));
+        let v_r = Some(virtual_register_index(1));
+        let v_qy = Some(virtual_register_index(2));
+
+        let mut virtual_sequence = vec![];
+
+        let q = ADVICEInstruction::<WORD_SIZE>(23434).lookup_entry();
+        virtual_sequence.push(RVTraceRow {
+            instruction: ELFInstruction {
+                address: trace_row.instruction.address,
+                opcode: RV32IM::VIRTUAL_ADVICE,
+                rs1: None,
+                rs2: None,
+                rd: trace_row.instruction.rd,
+                imm: None,
+                virtual_sequence_index: Some(0),
+            },
+            register_state: RegisterState {
+                rs1_val: None,
+                rs2_val: None,
+                rd_post_val: Some(q),
+            },
+            memory_state: None,
+            advice_value: Some(23434), // What should advice value be here?
```

Comment:
> What should advice value be here?

We know `x` and `y`, so we can just compute the quotient and remainder:

```rust
let q = x / y;
let r = x - q * y;
```

And `q` and `r` are the values we use for these two ADVICE instructions

Reply:
Got it, added!